crates/wadm-types/src/lib.rs

```diff
@@ -271,28 +400,42 @@ impl PartialEq<ConfigProperty> for String {
 
 /// Properties for links
 #[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+#[serde(deny_unknown_fields)]
 pub struct LinkProperty {
-    /// The target this link applies to. This should be the name of a component in the manifest
-    pub target: String,
     /// WIT namespace for the link
     pub namespace: String,
     /// WIT package for the link
     pub package: String,
     /// WIT interfaces for the link
     pub interfaces: Vec<String>,
     /// Configuration to apply to the source of the link
-    #[serde(default, skip_serializing_if = "Vec::is_empty")]
-    pub source_config: Vec<ConfigProperty>,
+    pub source: Option<ConfigDefinition>,
     /// Configuration to apply to the target of the link
-    #[serde(default, skip_serializing_if = "Vec::is_empty")]
-    pub target_config: Vec<ConfigProperty>,
```

Comment:
Went ahead and updated the code so that both `source_config` and `target_config` transformed into the new representation. We'll have to figure out how to prepare to remove them entirely.